 Keyguard: don't refresh ui when screen off

KeyguardStatusView is doing refresh all the time,
which cause high power when screen off.

diff --git a/packages/Keyguard/src/com/android/keyguard/KeyguardStatusView.java b/packages/Keyguard/src/com/android/keyguard/KeyguardStatusView.java
index 09002fa..1c95479 100644
--- a/packages/Keyguard/src/com/android/keyguard/KeyguardStatusView.java
+++ b/packages/Keyguard/src/com/android/keyguard/KeyguardStatusView.java
@@ -50,11 +50,18 @@
     private TextClock mClockView;
     private TextView mOwnerInfo;
 
+    //On the first boot, keyguard will start to receiver TIME_TICK intent.
+    //And onScreenTurnedOff will not get called if power off when keyguard is not started.
+    //Set initial value to false to skip the above case.
+    private boolean enableRefresh = false;
+
     private KeyguardUpdateMonitorCallback mInfoCallback = new KeyguardUpdateMonitorCallback() {
 
         @Override
         public void onTimeChanged() {
-            refresh();
+            if (enableRefresh) {
+                refresh();
+            }
         }
 
         @Override
@@ -69,11 +76,14 @@ public void onKeyguardVisibilityChanged(boolean showing) {
         @Override
         public void onStartedWakingUp() {
             setEnableMarquee(true);
+            enableRefresh = true;
+            refresh();
         }
 
         @Override
         public void onFinishedGoingToSleep(int why) {
             setEnableMarquee(false);
+            enableRefresh = false;
         }
 
         @Override
